pkg/pillar/cmd/volumemgr/volumemgr.go

```diff
@@ -696,3 +706,63 @@ func maybeUpdateConfigItems(ctx *volumemgrContext, newConfigItemValueMap *types.
 		}
 	}
 }
+
+//returns empty if not found
+func getZVolsByName(ctx *volumemgrContext, poolAndName string) string {
+	ctx.zfsVolMapMutex.Lock()
+	defer ctx.zfsVolMapMutex.Unlock()
+	for k, v := range ctx.zfsVolMap {
+		if v == poolAndName {
+			return k
+		}
+	}
+	return ""
+}
+
+func removeZVolsByDev(ctx *volumemgrContext, dev string) {
+	ctx.zfsVolMapMutex.Lock()
+	defer ctx.zfsVolMapMutex.Unlock()
+	delete(ctx.zfsVolMap, dev)
+}
+
+func removeZVolsByDataset(ctx *volumemgrContext, dataset string) {
+	ctx.zfsVolMapMutex.Lock()
+	defer ctx.zfsVolMapMutex.Unlock()
+	for k, v := range ctx.zfsVolMap {
+		if v == dataset {
+			delete(ctx.zfsVolMap, k)
+			break
+		}
+	}
+}
+
+//processZVols iterates over /dev/zd* devices and call ioctl to obtain pool/dataset from it
+func processZVols(ctx *volumemgrContext) {
+	searchPath := "/dev"
+	subitems, err := ioutil.ReadDir(searchPath)
+	if err != nil {
+		log.Errorf("processZVols ReadDir: %s", err)
+		return
+	}
+	ctx.zfsVolMapMutex.Lock()
+	defer ctx.zfsVolMapMutex.Unlock()
+	for _, subitem := range subitems {
+		if subitem.IsDir() {
+			continue
+		}
+		if !strings.HasPrefix(subitem.Name(), "zd") {
+			continue
+		}
+		devPath := filepath.Join(searchPath, subitem.Name())
+		if _, ok := ctx.zfsVolMap[devPath]; ok {
+			continue
+		}
+		poolAndDataset, err := zfs.GetPoolAndDatasetByDevice(log, devPath)
+		if err != nil {
+			log.Errorf("processZVols: %s", err)
+			continue
+		}
+		ctx.zfsVolMap[devPath] = poolAndDataset
```

Comment:
Is it a pool name or a poolAndDataset name? Confusing names of variables.

Reply:
I try to work with naming and put dataset as a more general variant of naming the things here